spec/api/comment_thread_spec.rb

```diff
@@ -88,60 +94,66 @@ def thread_result(params)
-          rs.length.should == 2 
+          rs, rs_count = thread_result course_id: "omg", group_id: 100, sort_order: "asc"
+          rs.length.should == 2
+          rs_count.should == 2
           rs.each_with_index { |res, i|
             check_thread_result_json(nil, @threads["t#{i+1}"], res)
             res["course_id"].should == "omg"
           }
         end
         it "returns an empty result when no threads match course_id" do
-          rs = thread_result course_id: 99
-          rs.length.should == 0 
+          rs, rs_count = thread_result course_id: 99
+          rs.length.should == 0
+          rs_count.should == 0
         end
         it "returns only group-less threads when no threads have matching group id" do
           @threads["t1"].group_id = 123
           @threads["t1"].save!
-          rs = thread_result course_id: DFLT_COURSE_ID, group_id: 321
+          rs, rs_count = thread_result course_id: DFLT_COURSE_ID, group_id: 321
+          rs_count.should == 9
           rs.each.map {|res| res["group_id"].should be_nil }
         end
         context "when filtering flagged posts" do
           it "returns threads that are flagged" do
             @threads["t1"].abuse_flaggers = [1]
             @threads["t1"].save!
-            rs = thread_result course_id: DFLT_COURSE_ID, flagged: true
-            rs.length.should == 1 
+            rs, rs_count = thread_result course_id: DFLT_COURSE_ID, flagged: true
+            rs.length.should == 1
+            rs_count.should == 1
             check_thread_result_json(nil, @threads["t1"], rs.first)
           end
           it "returns threads that have flagged comments" do
             @comments["t2 c3"].abuse_flaggers = [1]            
             @comments["t2 c3"].save!
-            rs = thread_result course_id: DFLT_COURSE_ID, flagged: true
-            rs.length.should == 1 
+            rs, rs_count = thread_result course_id: DFLT_COURSE_ID, flagged: true
+            rs.length.should == 1
+            rs_count.should == 1
             check_thread_result_json(nil, @threads["t2"], rs.first)
           end
           it "returns an empty result when no posts were flagged" do
-            rs = thread_result course_id: DFLT_COURSE_ID, flagged: true
-            rs.length.should == 0 
+            rs, rs_count = thread_result course_id: DFLT_COURSE_ID, flagged: true
+            rs.length.should == 0
+            rs_count.should == 0
           end
         end
         it "filters unread posts" do
           user = create_test_user(Random.new)
-          rs = thread_result course_id: DFLT_COURSE_ID, user_id: user.id
+          rs, rs_count = thread_result course_id: DFLT_COURSE_ID, user_id: user.id
           rs.length.should == 10
-          rs2 = thread_result course_id: DFLT_COURSE_ID, user_id: user.id, unread: true
+          rs2, rs2_count = thread_result course_id: DFLT_COURSE_ID, user_id: user.id, unread: true
```

Comment:
1. I personally don't like returning the tuple, especially when only 1 of the results is sometimes used.
2. If you leave it like this, Ned said you might use something like "__" (double underscore) to denote a throw away value.
3. It's odd that you have introduced unique names that aren't used.  Did you want assertions?  Or did you want these to be throw away?